src/utils/abiManager.js

```diff
@@ -0,0 +1,40 @@
+class ABIManager{
+  chainId = null;
+
+  constructor(chainId){
+      this.chainId = chainId;
+  }
+  
+  getABIDirectory(){
+    const ABIDirectory = __dirname +  '/../abis';
+    const DirectoryByChainID = {
+      "1": "/mainnet",
+      "4": "/rinkeby",
+      "31337": "/hardhat",
+    };
+    console.log(this.chainId)
+    const subDir = DirectoryByChainID[this.chainId];
+    console.log(ABIDirectory);
+    if (subDir !== undefined){
+      return ABIDirectory + subDir;
+    }
+    console.log(`chainId: ${this.chainId} was not in mapping`)
+    return ABIDirectory + '/hardhat';
+  }
+  
+  async KasuContract() {
+      return await import(this.getABIDirectory() + '/Kasu.json');
```

Comment:
might be problematic to use this, when the provider() is called its not `asynchronous` it might be calling too eagerly.

Reply:
I'm not sure if importing a dynamic file path works well this way, I tried a bit yesterday without much success.
If this doesn't work, you can always just import all known directories synchronously. (Add dummy files to make sure all imports are successful).

Reply:
Dude i have no idea why this was not working, i gave up on this approach.... the new way works it just ads a little helper file. It might be something with path availability after the client loads the index.js file.